jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -764,10 +761,8 @@ public final boolean isAlive() {
  * @see			Thread#isAlive
  */
 private boolean isDead() {
-	// Has already started, is not alive anymore, and has been removed from the ThreadGroup
-	synchronized(lock) {
-		return started && threadRef == NO_REF;
-	}
+	/* Has already started and is not alive anymore. */
+	return started && threadRef == NO_REF;
```

Comment:
suggest for readability and match with the extension repo versions:
`return (started && (threadRef == NO_REF))`